modules/vectra.py

```diff
@@ -3,19 +3,35 @@
 import warnings
 import html
 import sys
+import re
 
 warnings.filterwarnings('always', '.*', PendingDeprecationWarning)
 
+class HTTPException(Exception):
+    def __init__(self,status_code, response):
+        try: 
+            r = response.json()
+            if 'detail' in r:
+                detail = r['detail']
+            elif 'errors' in r:
+                detail = r['errors'][0]['title']
+            elif '_meta' in r:
+                detail = r['_meta']['message']
+            else:
+                detail = response.content
+        except: 
+            detail = response.content
+        body = 'Status code: {code} - {detail}'.format(code=str(status_code), detail=detail)
+        Exception.__init__(self, body)
```

Comment:
The correct way to call the parent class for python2/3 is:
super(HTTPException, self).__init__(body)

However, I don't think you need to call the parent class here.  Just set the message
instead of` super(HTTPException, self).__init__(body)` do `self.message = body`

Reply:
I think we need to initialize the parent class before we can set self.message. I'll go with the first suggestion.